File: backend/src/main/java/com/back/team11/domain/wishlist/dto/WishlistResponse.java
package com.back.team11.domain.wishlist.dto;

import com.back.team11.domain.wishlist.entity.Wishlist;

import java.time.LocalDateTime;

public record WishlistResponse(
        Long wishlistId,
        Long cafeId,
        String cafeName,
        LocalDateTime createAt
) {
    public static WishlistResponse from(Wishlist wishlist){
        return new WishlistResponse(
                wishlist.getId(),
                wishlist.getCafe().getId(),
                wishlist.getCafe().getName(),
                wishlist.getCreatedAt()
        );
    }
}

File: backend/src/main/java/com/back/team11/domain/wishlist/dto/WishlistResponse.kt
package com.back.team11.domain.wishlist.dto

import com.back.team11.domain.wishlist.entity.Wishlist
import java.time.LocalDateTime

data class WishlistResponse(
    val wishlistId: Long,
    val cafeId: Long,
    val cafeName: String,
    val createAt: LocalDateTime?
) {
    companion object {
        @JvmStatic
        fun from(wishlist: Wishlist): WishlistResponse {
            return WishlistResponse(
                wishlistId = wishlist.id,
                cafeId = wishlist.cafe?.id ?: 0,
                cafeName = wishlist.cafe?.name ?: "",
                createAt = wishlist.createdAt
            )
        }
    }
}
